script.js

```diff
@@ -1,41 +1,106 @@
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+// Function to initialize the webpage
+function initializePage() {
+  fetchEpisodesData();
 }
 
-function makePageForEpisodes(episodeList) {
+// Fetch episode data from TVMaze API
+function fetchEpisodesData() {
+  const apiUrl = 'https://api.tvmaze.com/shows/82/episodes';
+
+  fetch(apiUrl)
+      .then(response => {
+          if (!response.ok) {
+              throw new Error('Network response was not ok');
+          }
+          return response.json();
+      })
+      .then(episodesData => {
+          renderEpisodes(episodesData);
+          addSearchEventListener(episodesData);
+      })
+      .catch(error => {
+          handleFetchError(error);
+      });
+}
+
+// Render episodes on the webpage
+function renderEpisodes(episodesData) {
   const rootElem = document.getElementById("root");
+  removeAllChildren(rootElem);
+
+  episodesData.forEach(episode => {
+      const episodeCard = createEpisodeCard(episode);
+      rootElem.appendChild(episodeCard);
+  });
 
-  episodeList.forEach(episode => {
-      const episodeContainer = document.createElement("div");
-      episodeContainer.classList.add("episode");
+  displayEpisodeCount(episodesData.length);
+}
+
+// Create an episode card
+function createEpisodeCard(episode) {
+  const episodeContainer = document.createElement("div");
+  episodeContainer.classList.add("episode");
 
-      // Here I created the episode code
-      const episodeCode = `S${pad(episode.season)}E${pad(episode.number)}`;
+  const episodeCode = `S${pad(episode.season)}E${pad(episode.number)}`;
```

Comment:
You can create another function that returns the episode code to you and use it here. 
`function getEpisodeCode(episode)`